pkg/cli/demo.go

```diff
@@ -229,11 +229,15 @@ func checkDemoConfiguration(
 				return nil, errors.Newf("--nodes with a value different from 9 cannot be used with %s", geoFlag)
 			}
 		} else {
-			const msg = `#
+			demoCtx.nodes = 9
+			if !sqlCtx.embeddedMode {
+				// Only explain how the configuration was interpreted if the
+				// user has control over it.
+				const msg = `#
 # --geo-partitioned replicas operates on a 9 node cluster.
 # The cluster size has been changed from the default to 9 nodes.`
-			fmt.Println(msg)
-			demoCtx.nodes = 9
+				fmt.Println(msg)
```

Comment:
i do wonder if we should make a new function `PrintlnIfNotEmbedded` or something to simplify the if checks. but optional.

Reply:
I like the idea! I'll try it out.